pkg/action/mount.go

```diff
@@ -0,0 +1,318 @@
+/*
+Copyright © 2022 - 2023 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package action
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/joho/godotenv"
+
+	"github.com/rancher/elemental-toolkit/pkg/constants"
+	"github.com/rancher/elemental-toolkit/pkg/elemental"
+	v1 "github.com/rancher/elemental-toolkit/pkg/types/v1"
+	"github.com/rancher/elemental-toolkit/pkg/utils"
+)
+
+func RunMount(cfg *v1.RunConfig, spec *v1.MountSpec) error {
+	cfg.Logger.Info("Running mount command")
+
+	e := elemental.NewElemental(&cfg.Config)
+
+	cfg.Logger.Debugf("Mounting partitions")
+	err := e.MountPartitions(spec.Partitions.PartitionsByMountPoint(false))
+	if err != nil {
+		return err
+	}
+
+	cfg.Logger.Debugf("Mounting image %s", spec.Image.File)
+	if err := e.MountImage(spec.Image); err != nil {
+		cfg.Logger.Errorf("Error mounting image %s: %s", spec.Image.File, err.Error())
+		return err
+	}
+
+	if spec.RunCloudInit {
+		cfg.Logger.Debug("Running rootfs cloud-init stage")
+		err = utils.RunStage(&cfg.Config, "rootfs", cfg.Strict, cfg.CloudInitPaths...)
```

Comment:
This feels a little hacky since all the other stages are installed in the `elemental-setup` feature. Maybe we should move the units from that feature into elemental-rootfs and then try to `systemd-cat` this stage into the journal to get more streamlined handling of the different stages? Or maybe move the fs-dependant stages into the mount-command.